src/tests/DomainGuest.t.sol

```diff
@@ -363,33 +410,63 @@ contract DomainGuestTest is DSSTest {
         assertEq(vat.sin(address(guest)), 125 * RAD);
 
         // Will push out a deficit of 25 DAI (125 - 100)
-        guest.push();
+        guest.deficit();
 
+        assertEq(vat.dai(address(guest)), 100 * RAD);
+        assertEq(vat.sin(address(guest)), 125 * RAD);
+        guest.heal();
         assertEq(vat.dai(address(guest)), 0);
         assertEq(vat.sin(address(guest)), 25 * RAD);
         assertEq(guest.rid(), 1);
-        assertEq(guest.lastPayload(), abi.encodeWithSelector(DomainHostLike.push.selector, 0, -int256(25 ether)));
+        assertEq(guest.lastPayload(), abi.encodeWithSelector(DomainHostLike.deficit.selector, 0, 25 ether));
+    }
+
+    function testPushDeficitNoneExisting() public {
+        guest.file("dust", 100 ether);
+        vat.suck(address(this), address(guest), 101 * RAD);
+        vat.suck(address(guest), address(this), 100 * RAD);
+
+        assertEq(vat.dai(address(guest)), 101 * RAD);
+        assertEq(vat.sin(address(guest)), 100 * RAD);
+
+        vm.expectRevert("DomainGuest/non-deficit");
+        guest.deficit();
     }
 
     function testPushDeficitDust() public {
-        guest.file("dust", 101 * RAD);
+        assertEq(vat.dai(address(guest)), 0);
+        assertEq(vat.sin(address(guest)), 0);
+
+        vm.expectRevert("DomainGuest/non-deficit");
+        guest.deficit();
+
+        guest.file("dust", 101 ether);
         vat.suck(address(guest), address(this), 100 * RAD);
 
         assertEq(vat.dai(address(guest)), 0);
         assertEq(vat.sin(address(guest)), 100 * RAD);
 
         vm.expectRevert("DomainGuest/dust");
-        guest.push();
+        guest.deficit();
     }
```

Comment:
I would add a test case where a deficit is pushed out, but the `sin` of the guest is decreased simulating some revenue coming in and `guest.heal()` being called. This will make `vat.sin(address(this)) - _dsin * RAY` negative. As I said above, this should have a more explicit error.

Reply:
Done.